app/models/mixins/amazon_s3_authentication_mixin.rb

```diff
@@ -0,0 +1,86 @@
+module AmazonS3AuthenticationMixin
+  extend ActiveSupport::Concern
+
+  #
+  # Connections
+  #
+  def connect(options = {})
+    username = options[:username] || authentication_userid(options[:auth_type])
+    password = options[:password] || authentication_password(options[:auth_type])
+    service  = :S3
+    # Note: The hard-coded region will be removed after manageiq-ui-class implements region selection
+    region   = options[:region] || "us-east-1"
+
+    self.class.raw_connect(username, password, service, region)
+  end
+
+  def with_depot_connection(options = {})
```

Comment:
Is this mixin specific to depots? If not, where else will it be used?

Reply:
Right now, yes.  The code was modified from the code used for the provider but obviously we could not reference provider-specific code in the main repo.  Additionally I needed some changes (and reductions) in functionality.

Reply:
Okay, that makes more sense.

If this is really only meant for the `FileDepotS3` class then I think we should probably hide and simplify this even more rather than making it a first-class mixin.